src/pages/Mypage/Order/Order.js

```diff
@@ -1,7 +1,251 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import ReactDaumPost from 'react-daumpost-hook';
+import { API } from '../../../config';
+import { Card, CardMedia, TextField, Button } from '@mui/material';
+import * as S from './Order.style';
 
 const Order = () => {
-  return <div>Order</div>;
+  const [orderData, setOrderData] = useState({});
+  const [loading, setLoding] = useState(true);
+  const [addressData, setAddressData] = useState({
+    bidId: Number.MAX_SAFE_INTEGER,
+    phoneNumber: '',
+    street: '',
+    address: '',
+    zipcode: '',
+    price: Number.MAX_SAFE_INTEGER,
+  });
+  const { street, address, zipcode, phoneNumber } = addressData;
+  const {
+    imageUrl,
+    userName,
+    userEmail,
+    itemName,
+    authorName,
+    bidPrice,
+    shippingFee,
+    totalPrice,
+  } = orderData;
+  const orderPriceList = [
+    {
+      id: 1,
+      name: '낙찰금액',
+      낙찰금액: bidPrice,
+    },
+    {
+      id: 2,
+      name: '배송비',
+      배송비: shippingFee,
+    },
+    {
+      id: 3,
+      name: '부가세',
+      부가세: '0원',
+    },
+  ];
+
+  const handleAddressDatailChange = e => {
+    const { name, value } = e.target;
+
+    if (name === 'phoneNumber') {
+      const phoneNumber = value.replace(/-/g, '');
+      const regex = /^(\d{2,3})(\d{3,4})(\d{4})$/;
+      const matched = phoneNumber.match(regex);
+
+      if (matched) {
+        const formattedPhoneNumber = `${matched[1]}-${matched[2]}-${matched[3]}`;
+        setAddressData({ ...addressData, [name]: formattedPhoneNumber });
+      } else {
+        setAddressData({ ...addressData, [name]: phoneNumber });
+      }
+    } else {
+      setAddressData({ ...addressData, [name]: value });
+    }
+  };
+
+  const postConfig = {
+    onComplete: data => {
+      setAddressData(prev => ({ ...prev, address: data.address }));
+      setAddressData(prev => ({ ...prev, zipcode: data.zonecode }));
+    },
+  };
+
+  const postCode = ReactDaumPost(postConfig);
+
+  useEffect(() => {
+    fetch(API.ORDERS)
+      .then(res => res.json())
+      .then(({ orders: [data] }) => {
+        setOrderData(data);
+        setAddressData({
+          ...addressData,
+          bidId: data.bidId,
+          price: data.totalPrice,
+        });
+        setLoding(false);
+      });
+  }, []);
```

Comment:
```suggestion
const 000함수 =   fetch(API.ORDERS)
      .then(res => res.json())
      .then(({ orders: [data] }) => {
        setOrderData(data);
        setAddressData({
          ...addressData,
          bidId: data.bidId,
          price: data.totalPrice,
        });
        setLoding(false);
      });

  useEffect(() => {
  000함수()
  }, []);
```

함수 단위로 묶어서 함수 이름으로 어떻게 동작하는지 표현하면 
예상하기 좀더 수월 해 집니다.
반복되는 함수가 있다면 재사용하기도 좋구요.